action.yml

```diff
@@ -20,3 +26,71 @@ runs:
         file_path="$GITHUB_ACTION_PATH/settings.xml"
         
         echo "path=${file_path}" >> $GITHUB_OUTPUT
+
+    # PDI plugin integration test section
+    - name: Configure vm.max_map_count required by Elasticsearch
+      if: ${{ env.run_plugin_integration_tests }}
+      shell: bash
+      run: |
+        echo "vm.max_map_count=262144" >> /etc/sysctl.conf
+
+    - name: Run plugin integration tests
+      if: ${{ env.run_plugin_integration_tests && inputs.test_scenario_json && inputs.changed_modules }}
+      uses: lumada-common-services/gh-composite-actions@stable
+      env:
+        cmd_type: UNIT_TEST
```

Comment:
Agreed, this is not ideal.  What I really would like is an enhancement to the composite actions to add a new cmd type of "INTEGRATION_TEST". That way, the report could contain a separate entry for those kind of tests.

Here's a build where I triggered both unit and integration test failures: https://github.com/pentaho/pdi-plugins-ee/actions/runs/6906632202

Looks like the unit test reporting isn't working correctly - it shows a green arrow even though there were test failures.

Reply:
Here's a run where I introduced a unit test failure and set "fail on error" to true, so the integration tests didn't run at all.  We still get the green check box for unit tests in the report: https://github.com/pentaho/pdi-plugins-ee/actions/runs/6907982702/job/18796307926

Reply:
Here are some more examples: 
passing tests:  https://github.com/pentaho/pdi-plugins-ee/actions/runs/6950293652/job/18910153829
^ Doesn't look too bad as is, you just get two entries of "common-job-Unit-Test

integration test failure: https://github.com/pentaho/pdi-plugins-ee/actions/runs/6949504145/job/18907776948

unit failures - https://github.com/pentaho/pdi-plugins-ee/actions/runs/6950972398

Reply:
> We still get the green check box for unit tests in the repo

It's displaying a green mark because the unit test step was executed. Unit test tracking doesn't depend on whether the test passed or failed; it simply monitors whether the steps were executed.

https://github.com/pentaho/pdi-plugins-ee/actions/runs/6907982702/job/18796307926
In this case `Unit Test` step got executed. That why it showing green mark.

Reply:
That should probably be changed.  A green check mark next to "unit test" makes me think the tests passed, not just that they ran.   

And here, we're saying that green means "Successful", which again implies passing tests.
![report key](https://github.com/pentaho/actions-common/assets/69540820/74935777-d856-4f6a-91e6-9aa1af0aab04)

Reply:
Currently, the unit test report displays a success symbol if the unit test command is executed. We have created the ticket to work o  this.
https://hv-eng.atlassian.net/browse/HNC-507

Reply:
We have updated the composite action to display a 'boom 💥' symbol in the unit test step if the unit test command is successful but there are failed test cases.
Please re-run your workflows and verify the summary results.

Reply:
I'm not sure that that's running as expected.  I'm seeing that the symbol is dependent on the 'fail-on-error' setting: when 'fail-on-error' is set to true, you always get the green check mark.  When it's set to false, you get the 'boom' icon, but the overall build status shows as successful, even when there are test failures.  

Results:

- units passing, integration test failure, integration test fail-on-error=false
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7008073064/job/19063941589
the report shows the correct 'boom' icon, but the overall build status looks like it passed; not good

- units passing, integration test failures, integration test fail-on-error= true
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7008258200
the report shows the incorrect green mark icon, but the overall build status is failed; better

- units failing, fail on error=true
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7008813273
^ incorrect green check mark, overall build status is failed

- units failing, fail on error=false
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7009035472
^ correct boom icon, overall build status looks successful; not good

Reply:
This PR sets all 'fail-on-error' settings to true, which yields the more desirable behavior of marking the entire build failed in the case of test failures. 

Can this PR be merged?

Reply:
@bconlon-hv Sorry for the late reply. Having multiple steps with the same `cmd_type` is something that we haven't foreseen when started implementing the composite action. And because the usage of `UNIT_TEST` to run integration tests just to get the reporting functionality of it is conceptually incorrect (as expressed by you even), we have created https://hv-eng.atlassian.net/browse/HNC-510 to set a new `INTEGRATION_TEST` type to accommodate this use case. Feel free to add details if suitable.

Reply:
> I'm not sure that that's running as expected. I'm seeing that the symbol is dependent on the 'fail-on-error' setting: when 'fail-on-error' is set to true, you always get the green check mark. When it's set to false, you get the 'boom' icon, but the overall build status shows as successful, even when there are test failures.
> 
> Results:
> 
> * units passing, integration test failure, integration test fail-on-error=false
>   https://github.com/pentaho/pdi-plugins-ee/actions/runs/7008073064/job/19063941589
>   the report shows the correct 'boom' icon, but the overall build status looks like it passed; not good
> * units passing, integration test failures, integration test fail-on-error= true
>   https://github.com/pentaho/pdi-plugins-ee/actions/runs/7008258200
>   the report shows the incorrect green mark icon, but the overall build status is failed; better
> * units failing, fail on error=true
>   https://github.com/pentaho/pdi-plugins-ee/actions/runs/7008813273
>   ^ incorrect green check mark, overall build status is failed
> * units failing, fail on error=false
>   https://github.com/pentaho/pdi-plugins-ee/actions/runs/7009035472
>   ^ correct boom icon, overall build status looks successful; not good

@bconlon-hv 
1) units failing, fail on error=true
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7008813273
^ incorrect green check mark, overall build status is failed

-->We have updated our composite action to handle this scenario. Please rerun this specific use case to verify the result

2) units failing, fail on error=false
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7009035472
^ correct boom icon, overall build status looks successful; not good

-->Yes, it will show the 'boom💥' symbol, but the overall status will be marked as successful because we are passing `fail-on-error=false`, which marks the unit test step as successful even if there are test failures.

Although both conditions above will work perfectly when running only unit tests, if unit tests and integration tests are run under the same command (type UNIT_TEST), results may vary. We have created a ticket to accommodate this use case.

Reply:
@cardosov Ok, I will update the PR with the new cmd type when it becomes available

Reply:
@bconlon-hv @AdityaHitachi opened https://github.com/lumada-common-services/gh-composite-actions/pull/132 to address the new `cmd_type`. Maybe you can change your composite action references to point to [HNC-510](https://github.com/lumada-common-services/gh-composite-actions/tree/HNC-510) and change the `cmd_type` to `INTEGRATION_TEST` to check if this covers your use cases? Thanks

Reply:
@AdityaHitachi @cardosov 
I ran a few builds with cmd_type=INTEGRATION_TEST against HNC-510:
- all tests pass - https://github.com/pentaho/pdi-plugins-ee/actions/runs/7034627851
- units failing, fail-on-error=true
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7034830742
^ cool, looks like the icon issue has been fixed
- units passing, integration tests failing, fail-on-error=true
https://github.com/pentaho/pdi-plugins-ee/actions/runs/7034896336

Looks ok so far, but I'd like the integration tests to get an entry in the report summary table.  ^ In that last build, the integration tests and overall build failed, but the summary doesn't reflect that:
![no integration test](https://github.com/pentaho/actions-common/assets/69540820/daeb057a-d88a-4268-a62c-56a2ef28edc2)

(It's also important that the table include a separate entry for the integration tests because that table gets pasted into the slack notifications.)

Reply:
@bconlon-hv 
We have merged the pull request for integration test changes. Now, the `stable` tag points to the latest changes.

Reply:
@bconlon-hv The `cmd_type` can be changed to `INTEGRATION_TEST` now.

Reply:
Updated